File: 1031. Hello World for U (20).py
#返回题目要求的bot和side
def bot_side(N):
    for bot in range(N):
        side = (N - bot)//2 + 1
        if(bot+side*2 == N+2 and bot>=side):
            return bot,side,bot-2
if __name__ == '__main__':
    string = input("")
    N = len(string)
    bot,side,space = bot_side(N)
    #输出图形
    for i in range(side-1):
        print(string[i]+" "*space+string[-1-i])
    print(string[i+1:-i-1])

File: 1031. Hello World for U (20).cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
	string s;
	cin >> s;
	int n = s.length();
	int bot = 0, side = 0, flag = 0;
	for (int i=0; i<n;i++) {
		for(int j=0;j<n;j++) {
			side = (n-j)/2+1;
			if(j+side*2-2==n && j>=side) {
				bot = j;
				flag = 1;
				break;
			} 
		}
		if(flag==1) break;
	}

	int space = bot - 2;
	for(int i=0;i<side-1;i++){
		cout << s[i];
		for(int j=0;j<space;j++) {
			cout << " ";
		}
		cout << s[n-1-i] << endl;
	}
	cout << s.substr(side-1,bot) << endl;
	return 0;
}

